### PR TITLE
Support for de/serializing remaining crucial AstNodes

### PR DESCRIPTION
**Pull request by noahlemen**
doing this in one PR to save some time!
- [x] `NaryOperation`
- [x] `Conditional`
- [x] `TryCatchStatement`
- [x] `RegExpLiteral`
- [x] `SwitchStatement`
- [x] `Throw`

### CODE REVIEW

File: src/parsing/binast-deserializer.cc [hunk 1]
```diff
@@ -938,7 +956,7 @@ BinAstDeserializer::DeserializeResult<ObjectLiteral*> BinAstDeserializer::Deseri
 
   bool has_rest_property = false;
   ObjectLiteral* result = parser_->factory()->NewObjectLiteral(
-      properties, number_of_boilerplate_properties, literal_position.value,
```
Comment (mhahnenberg) [inline src/parsing/binast-deserializer.cc:941]: Was this a bug before?

File: src/parsing/binast-serialize-visitor.h [hunk 2]
```diff
@@ -918,13 +928,14 @@ inline void BinAstSerializeVisitor::VisitThisExpression(ThisExpression* this_exp
...
-  ToDoBinAst(reg_exp_literal);
+  SerializeRawStringReference(reg_exp_literal->raw_pattern());
+  SerializeInt32(reg_exp_literal->flags());
 }
 
 inline void BinAstSerializeVisitor::VisitSwitchStatement(
     SwitchStatement* switch_statement) {
   SerializeAstNodeHeader(switch_statement);
-  ToDoBinAst(switch_statement);
+  VisitNode(switch_statement->tag());
```
Comment (mhahnenberg) [inline src/parsing/binast-serialize-visitor.h:953]: What about all the cases for the switch statement? Might also be worth an example in test.js

**Review by mhahnenberg — CHANGES_REQUESTED** [review]
Everything looks solid except for SwitchStatement, which I have a question on.

File: src/parsing/binast-deserializer.cc (hunk 1 above)
Comment (noahlemen) [reply src/parsing/binast-deserializer.cc:941]: i don't think it was behaving incorrectly or anything, i just hadn't realized we already had position for it and i didn't need to serialize it again myself

File: src/parsing/binast-serialize-visitor.h (hunk 2 above)
Comment (noahlemen) [reply src/parsing/binast-serialize-visitor.h:953]: updated. briefly tried to get BreakStatement working, but handling `target` is a bit weird. putting a pin in that for now.

**Review by mhahnenberg — APPROVED** [review]
Nice!

**Pull request merged**